### PR TITLE
analyze: AST_to_IL: Translate most (all?) lvalue expressions

### PR DESCRIPTION
**Pull request by IagoAbal**
This is a WIP, just FYI, so that you know what I'm working on. But feel free to comment already if you see something wrong. I'll complete it ASAP.

This patch should translate most (all?) kinds of lvalue expressions from Generic AST to IL.

### CODE REVIEW

**Comment by IagoAbal** [comment]
Do we have a set of tests for AST-to-IL translation or we test this indirectly via Semgrep?

**Comment by IagoAbal** [comment]
Now it's good for review!

**Comment by aryx** [comment]
There's no test yet I think. I just use pfff -il_generic <file>.

File: lang_GENERIC/analyze/AST_to_IL.ml [hunk 1]
```diff
@@ -227,12 +239,9 @@ and pattern_assign_statements env exp eorig pat =
 (* Assign *)
 (*****************************************************************************)
 and assign env lhs _tok rhs_exp eorig =
-  match lhs with
-  | G.Id (_, _) ->
-      let lval = lval env lhs in
-      add_instr env (mk_i (Assign (lval, rhs_exp)) eorig);
-      mk_e (Lvalue lval) lhs
-  | _ -> todo (G.E lhs)
+  let lval = lval env lhs in
```
Comment (aryx) [inline lang_GENERIC/analyze/AST_to_IL.ml:242]: actually lhs can be a tuple, dict, or even object in languages such as JS that supports
destructuring assignment. We should "compile" those in simpler assign.

**Review by aryx — APPROVED** [review]

File: lang_GENERIC/analyze/AST_to_IL.ml (hunk 1 above)
Comment (IagoAbal) [reply lang_GENERIC/analyze/AST_to_IL.ml:242]: You're right, thanks for pointing it out!

**Pull request merged**